pyrasterframes/src/main/python/docs/languages.pymd

```diff
@@ -0,0 +1,160 @@
+
+```python, imports, echo=False
+from pyspark.sql.functions import *
+from pyrasterframes.utils import create_rf_spark_session
+
+from pyrasterframes.rasterfunctions import *
+import pyrasterframes.rf_ipython
+import pandas as pd
+import os
+spark = create_rf_spark_session()
+```
+
+## Python
+
+### Step 1: Load the catalog
+
+```python, step_1_python
+modis = spark.read.format('aws-pds-modis-catalog').load()
+```
+### Step 2: Down-select data by month
+
+```python, step_2_python
+red_nir_monthly_2017 = modis \
+    .select(
+        col('granule_id'), 
+        month('acquisition_date').alias('month'), 
+        col('B01').alias('red'), 
+        col('B02').alias('nir')
+    ) \
+    .where((year('acquisition_date') == 2017) & (dayofmonth('acquisition_date') == 15) & (col('granule_id') == 'h21v09'))
+```
+
+### Step 3: Read tiles
+
+```python, step_3_python
+red_nir_tiles_monthly_2017 = spark.read.raster(catalog=red_nir_monthly_2017, catalog_col_names=['red', 'nir'])
+```
+
+### Step 4: Compute aggregates
+
+```python, step_4_python
+result = red_nir_tiles_monthly_2017 \
+    .where(st_intersects(
+        st_reproject(rf_geometry(col('red')), rf_crs(col('red')).crsProj4, rf_mk_crs('EPSG:4326')), 
+        st_makePoint(lit(34.870605), lit(-4.729727)))
+    ) \
+    .groupBy('month') \
+    .agg(rf_agg_stats(rf_normalized_difference(col('nir'), col('red'))).alias('ndvi_stats')) \
+    .orderBy(col('month')) \
+    .select('month', 'ndvi_stats.*')
+result.show()
+```
+
+## SQL
+
+For convenience we're going to evaluate SQL from the Python environment. The SQL fragments should work in the `spark-sql` shell just the same.
+
+```python, sql_setup
+def sql(stmt):
+    return spark.sql(stmt)
+```
+
+### Step 1: Load the catalog
+
+```python, step_1_sql
+sql("CREATE OR REPLACE TEMPORARY VIEW modis USING `aws-pds-modis-catalog`")
+```
+
+### Step 2: Down-select data by month
+
+```python, step_2_sql
+sql("""
+CREATE OR REPLACE TEMPORARY VIEW red_nir_monthly_2017 AS
+SELECT granule_id, month(acquisition_date) as month, B01 as red, B02 as nir
+FROM modis
+WHERE year(acquisition_date) = 2017 AND day(acquisition_date) = 15 AND granule_id = 'h21v09'
+""")
+sql('DESCRIBE red_nir_monthly_2017').show() 
+```
+
+### Step 3: Read tiles
+
+```python, step_3_sql
+sql("""
+CREATE OR REPLACE TEMPORARY VIEW red_nir_tiles_monthly_2017 
+USING raster 
+OPTIONS (catalogTable='red_nir_monthly_2017', catalogColumns='red,nir')
+""")
+```
+
+### Step 4: Compute aggregates
+
+```python, step_4_sql
+sql("""
+SELECT month, ndvi_stats.* FROM (
+    SELECT month, rf_agg_stats(rf_normalized_difference(nir, red)) as ndvi_stats
+    FROM red_nir_tiles_monthly_2017
+    WHERE st_intersects(st_reproject(rf_geometry(red), rf_crs(red), 'EPSG:4326'), st_makePoint(34.870605, -4.729727))
+    GROUP BY month
+    ORDER BY month
+)
+""").show()
```

Comment:
I stared at the code a while. The show outputs here different between SQL and Python. I don't know why. 

It may be simpler to have count (row count), datacells, and mean.

Looks like the SQL has a lot more rows. or at least cells. I kind of wonder if there is some issue with using the same view names?